### PR TITLE
feat(j-s): Send additional data to court

### PR DESCRIPTION
**Pull request by oddsson**
# Send additional data to court

[Asana](https://app.asana.com/0/1199153462262248/1208055116618175/f)

## What

Enable defenders and prosecutors to send additional data to court. 

## Why

So that they can submit additional data to the court electronically and not send physical paper to the court.

### Upload fails

https://github.com/user-attachments/assets/562cc702-fa47-42f6-a4b6-eaa59fba7b4b

### One file fails, one doesn't

https://github.com/user-attachments/assets/2dceb5a9-2acb-4dcd-b69c-b0578d43739b

### Success

https://github.com/user-attachments/assets/bcee9a18-06b4-4d5e-b7a0-4a376bf0ecf1

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
  - Introduced a user-defined filename option when creating file entries.
  - Added a new button for adding documents in the indictment overview.
  - Implemented a file upload interface with drag-and-drop support for PDF files.
  - Enhanced the user interface with localized strings for the "Add Files" feature.

- **Bug Fixes**
  - Enhanced error handling for document access failures with user-friendly notifications.

- **Refactor**
  - Improved component structure and modularity for better maintainability.
  - Streamlined file upload process and error handling in the upload system.

- **Style**
  - Updated styles for a modern and responsive user interface in file management components.

### CODE REVIEW

File: apps/judicial-system/web/src/components/AccordionItems/IndictmentsCaseFilesAccordionItem/IndictmentsCaseFilesAccordionItem.tsx [hunk 1]
```diff
@@ -80,7 +74,7 @@ const useRaisedShadow = (value: MotionValue<number>) => {
 
   useEffect(() => {
     let isActive = false
-    value.onChange((latest) => {
```
Comment (oddsson) [inline apps/judicial-system/web/src/components/AccordionItems/IndictmentsCaseFilesAccordionItem/IndictmentsCaseFilesAccordionItem.tsx:83]: The reason for this change is that this syntax is deprecated.

File: apps/judicial-system/web/src/routes/Shared/AddFiles/AddFiles.tsx [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+  const handleFileUpload = useCallback(
+    async (files: UploadFile[]) => {
+      const filesUploaded = await handleUpload(files, updateFileToUpload)
+
+      if (filesUploaded) {
+        setVisibleModal('sendFiles')
+      }
+    },
+    [handleUpload, updateFileToUpload],
+  )
```
[the remark this answers is not in the document]
Comment (oddsson) [reply apps/judicial-system/web/src/routes/Shared/AddFiles/AddFiles.tsx:67]: All error handling is done in the `handleUpload` function.

File: apps/judicial-system/web/src/routes/Shared/AddFiles/AddFiles.tsx [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+      }
+
+      if (newDate) {
+        filesToUpload[fileToUpdate].displayDate = newDate.toISOString()
+      }
+
+      updateFileToUpload(uploadFiles[fileToUpdate])
+    },
+    [filesToUpload, formatMessage, updateFileToUpload, uploadFiles],
+  )
```
[the remark this answers is not in the document]
Comment (oddsson) [reply apps/judicial-system/web/src/routes/Shared/AddFiles/AddFiles.tsx:103]: UpdateFileToUpload is just a state change and is really unlikely to fail. Not doing this.

**Review by gudjong — COMMENTED** [review]
This is mostly done.

Two things remaining as far as I know:
- File open does not work for files that have not been uploaded. Not sure how to fix or if we want to fix it.
- The new file categories need to be added to robot emails.

**Comment by oddsson** [comment]
@gudjong, the two things you mentioned are stand-alone issues that have been added to our inbox. This was decided with Gudny IRL.

**Review by unakb — APPROVED** [review]

**Pull request merged**